.github/actions/pre-commit-zsh/action.yml

```diff
@@ -0,0 +1,21 @@
+name: "Test the pre-commit shell scripts in zsh"
+description: "Only works when ran on macos"
```

Comment:
isn't the whole idea that it should be compatible cross-platform?

Reply:
Unfortunately, no. Perhaps we could install zsh in the ubuntu tests as a step?

Reply:
zsh comes preinstalled only on mac, not on the ubuntu runner, hence if you run the action on ubuntu it will return an error

Reply:
https://github.com/actions/runner-images/issues/264